lightning/src/routing/router.rs

```diff
@@ -343,8 +343,8 @@ fn compute_fees(amount_msat: u64, channel_fees: RoutingFees) -> Option<u64> {
 /// The fees on channels from us to next-hops are ignored (as they are assumed to all be
 /// equal), however the enabled/disabled bit on such channels as well as the
 /// htlc_minimum_msat/htlc_maximum_msat *are* checked as they may change based on the receiving node.
-pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, payee: &PublicKey, payee_features: Option<InvoiceFeatures>, first_hops: Option<&[&ChannelDetails]>,
-	last_hops: &[&RouteHintHop], final_value_msat: u64, final_cltv: u32, logger: L) -> Result<Route, LightningError> where L::Target: Logger {
+pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, payee: &PublicKey, payee_features: Option<InvoiceFeatures>, first_hops: Option<&[ChannelDetails]>,
+	last_hops: Vec<&[RouteHintHop]>, final_value_msat: u64, final_cltv: u32, logger: L) -> Result<Route, LightningError> where L::Target: Logger {
```

Comment:
Primarily because `RouteHint`'s interface uses an error type from `lightning-invoice`, and I didn't want to go down the path of changing that. Also note that `Invoice`'s interface gives `Vec<&RouteHint>` where `RouteHint` wraps `Vec< RouteHintHop>`.

Reply:
Hmm, can we just swap CreationError for ()? We only fail in one case and it's easy enough to map it on the invoice side.

Reply:
Considered doing this, but the check really belongs in `lightning-invoice` as it has to do with keeping the route within a serialization limitation.

https://github.com/rust-bitcoin/rust-lightning/blob/a8038a8234b70bc261f633a9e2b40adebebcd8d4/lightning-invoice/src/lib.rs#L419

Instead, I made a `RouteHint` struct in `lightning` and wrapped it in `lightning-invoice` by `Route` (renaming `RouteHint` there), which is a more consistent name with the other fields. Ended up adding a `route_hints` method to `Invoice` to do the conversion. Was hoping there was a simple way to do this with `Into` but couldn't quite get something working.